docs/icd.md

```diff
@@ -0,0 +1,195 @@
+
+[`gRPC`](https://grpc.io/) is a modern, open-source Remote Procedure Call (RPC) framework.
+
+Features of gRPC:
++ Serializes data into Protocol Buffers, strongly typed messages
++ Integrated TLS/SSL authentication and encryption
++ Supports both client and server-side streaming.
+
+Arrow services, which are written in Rust Programming Language, access gRPC features through the [`tonic`](https://docs.rs/tonic/latest/tonic/) library.
+
+### :scroll: Files
+
+A `lib-common` repository will contain gRPC interfaces that will be used by every microservice.
+
+File | Description
+---- | -----
+`common-grpc.proto` | Common gRPC interfaces, protocol buffer file.
+`common-grpc.rs` | Autogenerated Rust Code
+
+The `lib-common` crate will be published to a registry, and added as a dependency by each service:
+```toml
+# e.g. svc-scheduler Cargo.toml
+[dependencies]
+lib-common = 0.1.2
+```
+
+Service-specific gRPC files will be defined in each service's repository:
+```tree
+.
+├── Cargo.lock
+├── Cargo.toml # [workspace] members = ['client', 'server']
+├── Makefile
+├── client/
+│   ├── Cargo.lock
+│   ├── Cargo.toml
+│   └── src
+│       ├── grpc.rs # autogenerated during `cargo build`
+│       └── lib.rs
+├── proto/
+│   └── grpc.proto # used to autogenerate server and client stubs
+└── server
+    ├── build.rs
+    ├── Cargo.toml
+    ├── src
+    │   ├── grpc.rs # autogenerated during `cargo build`
+    │   ├── main.rs
+```
+Details:
+- A service codebase shall contain two Rust crates, defined in the directories `server` and `client`.
+  - The `server` crate is an executable binary
+  - The `client` crate is a library
+- There shall be a `proto/` directory containing protocol buffer files.
+- During `cargo build`, the protobuf files will be used to autogenerate gRPC server and client stubs.
+  - The autogenerated server code is created in the `server/src` directory
+  - The autogenerated client code is created in the `client/src` directory.
+- Each crate is then compiled
+- CI/CD will separately publish these crates to a registry.
+  - `svc-scheduler` will publish `svc-scheduler` (server binary) and `svc-scheduler-client` (client library).
+
+A client will access a server's structures and functions by adding a dependency for the target's client crate:
+```toml
+# svc-pricing (consumer) Cargo.toml 
+[dependencies]
+svc-scheduler-client = 0.1.0
+```
+### :space_invader: Integrated Authentication & Encryption
+
+gRPC supports Transport Layer Security (TLS): https://grpc.io/docs/guides/auth/
+
+SSL/TLS certificates will be stored as `.pem` file contents in the GitHub organization as [Encrypted Environment Secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets). These environment secrets can be accessed by the CI/CD pipeline without exposing the information to the public.
+
```

Comment:
I think we might actually store these in AWS SSM Parameter store and/or in GCP Secrets manager since we'll need to have access to the certs when running the services in the cloud.
The CICD pipeline won't need access if we run the integration tests on VM's in AWS or GCP (which we most likely will).
These VM's are spun up with specific Service Accounts which will have access to pull in the certificates from the right secret store. This can be exposed to the dockers running on the VM's through a [docker secret](https://docs.docker.com/engine/swarm/secrets/) parameter

Reply:
Added that as a possibility: https://github.com/Arrow-air/se-services/pull/3/commits/53bbe5a7abe6b62682e2b655bb951f3423d7beca